File: spring-web-java/src/main/java/spring/web/java/presentation/dto/request/OrderSaveRequest.java
package spring.web.java.presentation.dto.request;

import java.util.List;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record OrderSaveRequest(
	@NotNull
	Long memberId,
	@NotBlank
	String city,
	@NotBlank
	String street,
	@NotBlank
	String zipcode,
	@Size(min = 1)
	List<OrderItemSaveRequest> orderItems
) {
	public OrderSaveRequest {
		if (orderItems == null) {
			orderItems = List.of();
		}
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/request/OrderSaveRequest.kt
package spring.web.kotlin.presentation.dto.request

import jakarta.validation.constraints.Size

data class OrderSaveRequest(
    val memberId: Long,
    val city: String,
    val street: String,
    val zipcode: String,
    @field:Size(min = 1)
    val orderItems: List<OrderItemSaveRequest> = emptyList(),
)
